zap/src/irgen/des.rs

```diff
@@ -86,34 +86,36 @@ impl Des {
 				if let Some(len) = range.exact() {
 					self.push_assign(into, self.readstring(len.into()));
 				} else {
-					self.push_local("len", Some(self.readnumty(NumTy::U16)));
+					let len_name = format!("len_{}", self.buf.len());
```

Comment:
Why did you go with the string length instead of just counting up each time `len` is used for example? It could be a HashMap or something in `irgen`?